src/webserver.cpp

```diff
@@ -601,6 +601,15 @@ MHD_Result webserver::finalize_answer(MHD_Connection* connection, struct details
                 }
             } else {
                 mr->dhrs = method_not_allowed_page(mr);
+
+                vector<string> allowed_methods = hrm->get_allowed_methods();
+                if (allowed_methods.size() > 0) {
+                    string header_value = allowed_methods[0];
+                    for (auto it = allowed_methods.cbegin() + 1; it != allowed_methods.cend(); ++it) {
+                        header_value += ", " + (*it);
+                    }
+                    mr->dhrs->with_header("Allow", header_value);
```

Comment:
Can we use: https://github.com/etr/libhttpserver/blob/263f2bdef4d5f8887a46c19fa08695170ad679f7/src/http_utils.cpp#L135

Reply:
Sure, no problem. Force pushed the updated branch.